### PR TITLE
feat: ftp server related updates

### PR DESCRIPTION
**Pull request by bvanderhorn**
- ProblemDetailsException
- FtpServerHost
- String generation with Nanoid
- HttpHeadersExtensions => AddCorrelationId

### CODE REVIEW

File: src/ShipitSmarter.Core.AspNet/Extensions/HttpHeadersExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using System.Net.Http.Headers;
+
+namespace ShipitSmarter.Core.AspNet.Extensions;
```
Comment (ThumNet) [inline src/ShipitSmarter.Core.AspNet/Extensions/HttpHeadersExtensions.cs:3]: Use the `System.Net.Http.Headers` as namespace, this makes the extension method available without adding an extra using.

You can you the disable once with comment, to make the compiler happy

File: src/ShipitSmarter.Core/Exceptions/ProblemDetailsException.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    /// <summary>
+    /// Instantiates a new instance of a <exception cref="ProblemDetailsException"></exception>
+    /// </summary>
+    /// <param name="problemDetails">The problem details to be used</param>
+    /// <param name="message">(Optional) The message that describes the error.</param>
+    public ProblemDetailsException(CoreProblemDetails problemDetails, string? message = null) 
+        : base(message ?? problemDetails.Title ?? "An error occurred")
+    {
+        Detail = problemDetails.Detail;
+        StatusCode = problemDetails.Status ?? 400;
```
Comment (ThumNet) [inline src/ShipitSmarter.Core/Exceptions/ProblemDetailsException.cs:20]: Should the default not be a 500?
400 ranges are for client side errors

**Review by ThumNet — CHANGES_REQUESTED** [review]
Nice, small remarks

File: src/ShipitSmarter.Core.AspNet/Extensions/HttpHeadersExtensions.cs (hunk 1 above)
Comment (bvanderhorn) [reply src/ShipitSmarter.Core.AspNet/Extensions/HttpHeadersExtensions.cs]: agreed

File: src/ShipitSmarter.Core/Exceptions/ProblemDetailsException.cs (hunk 2 above)
Comment (bvanderhorn) [reply src/ShipitSmarter.Core/Exceptions/ProblemDetailsException.cs]: agreed

**Review by ThumNet — APPROVED** [review]

**Pull request merged**